networking/firmware_protocol/src/lib.rs

```diff
@@ -90,17 +77,51 @@ impl SlimeString {
 }
 
 #[derive(Debug, PartialEq, DekuRead, DekuWrite)]
-#[deku(type = "u32")]
 #[deku(endian = "big")]
+pub struct Packet {
+	// TODO: This tag could really be dropped from the Rust side, but #[deku(temp)] is a bit wonky
+	tag: u32,
+	seq: u64,
+	#[deku(ctx = "*tag")]
+	ty_: PacketType,
+}
+
+impl Packet {
+	pub fn new(seq: u64, ty_: PacketType) -> Packet {
+		Packet {
+			tag: ty_.deku_id().unwrap(),
+			seq,
+			ty_,
+		}
+	}
+
+	pub fn serialize_into(&self, buf: &mut [u8]) -> usize {
+		let bytes = self.to_bytes().unwrap();
+		buf[..bytes.len()].copy_from_slice(&bytes);
+		bytes.len()
+	}
+
+	pub fn deserialize_from(buf: &[u8]) -> Result<Packet, ()> {
+		match Packet::from_bytes((buf, 0)) {
+			Ok(((tail, _), p)) if tail.len() == 0 => Ok(p),
+			_ => Err(()),
+		}
+	}
+
+	pub fn split(self) -> (u64, PacketType) {
+		(self.seq, self.ty_)
+	}
+}
+
+#[derive(Debug, PartialEq, DekuRead, DekuWrite)]
+#[deku(ctx = "e: deku::ctx::Endian, tag: u32", id = "tag", endian = "e")]
```

Comment:
Inheriting the endian from the deku context seems a bit of an overcomplication. I know its "more correct" but the ctx syntax is a bit of a DSL that I would like to avoid. 

I'm inclined to suggest we just hard code it as big endian, thereby simplifying the macro syntax. Thoughts?

Reply:
Because of how deku ctx works, its not really simpler but I agree, endianness should be hardcoded to big